client/my-sites/media/main.jsx

```diff
@@ -97,54 +115,141 @@ export default React.createClass( {
 
 		MediaActions.update( site.ID, item, true );
 		resetAllImageEditorState();
-		this.setState( { openedDetails: null, editedItem: null } );
+		this.setState( { currentDetail: null, editedItem: null, selectedImages: [] } );
 	},
+
+	getModalButtons() {
+		return [
+			{
+				action: 'delete',
+				additionalClassNames: 'media__modal-delete-item-button is-link',
+				label: this.translate( 'Delete' ),
+				isPrimary: false,
+				disabled: false,
+				onClick: this.deleteMediaByItemDetail,
+			},
+			{
+				action: 'confirm',
+				label: this.translate( 'Done' ),
+				isPrimary: true,
+				disabled: false,
+				onClose: this.closeDetailsModal,
+			}
+		];
+	},
+
 	restoreOriginalMedia: function( siteId, item ) {
 		if ( ! siteId || ! item ) {
 			return;
 		}
 		MediaActions.update( siteId, { ID: item.ID, media_url: item.guid }, true );
-		this.setState( { openedDetails: null, editedItem: null } );
+		this.setState( { currentDetail: null, editedItem: null, selectedImages: [] } );
+	},
+
+	setDetailSelectedIndex: function( index ) {
+		this.setState( { currentDetail: index } );
+	},
+
+	/**
+	 * Start the process to delete media items.
+	 * `callback` is an optional parameter which will execute once the confirm dialog is accepted.
+	 * It's used especially when the item is attempting to be removed using the item detail dialog.
+	 *
+	 * @param  {Function} [callback] - callback function
+	 */
+	deleteMedia: function( callback ) {
+		const site = this.props.sites.getSelectedSite();
+		const selected = MediaLibrarySelectedStore.getAll( site.ID );
+		const selectedCount = selected.length;
+		const confirmMessage = this.translate(
+			'Are you sure you want to permanently delete this item?',
+			'Are you sure you want to permanently delete these items?',
+			{ count: selectedCount }
+		);
+
+		accept( confirmMessage, accepted => {
+			if ( ! accepted ) {
+				return null;
```

Comment:
> You're right. Another reason to return null ;-)

the way I typically do it is that if my only reason to use `return` is to early-abort from a function then I'll just `return` because that's how it would have happened if the function had returned normally given no explicit value.

if, on the other hand, the function is designed to return a value and something else is counting on it I'll want to explicitly return something. bonus points if it's the same data type as would otherwise return.

for example, an empty string could be more appropriate for a string function than `null`, a `0` might be more appropriate for a number function, an empty list for a list function, etc…

but if I'm calling a function for its side-effects only and don't expect a value, I think getting _something_ back could also be confusing